src/Spryker/Zed/EventBehavior/Business/Model/TriggerManager.php

```diff
@@ -98,14 +98,14 @@ public function triggerRuntimeEvents()
             return;
         }
 
-        $events = $this->queryContainer->queryEntityChange($processId)->find()->getData();
-        static::$eventBehaviorTableExists = true;
-
-        $triggeredRows = $this->triggerEvents($events);
-
-        if ($triggeredRows !== 0 && count($events) === $triggeredRows) {
-            $this->queryContainer->queryEntityChange($processId)->delete();
-        }
+        $limit = $this->config->getTriggerChunkSize();
+        do {
+            $events = $this->queryContainer->queryEntityChange($processId)->find()->getData();
+            static::$eventBehaviorTableExists = true;
+            $countEvents = count($events);
+
+            $this->triggerEventsAndDelete($events);
+        } while ($countEvents === $limit);
```

Comment:
looks like a error to me. What if we have more events? I don't see any restricting on the query

Reply:
I testing and all good 

1. when we have maximum entities by limits we run loop interaction again 
2. If we have count < limit this is the last loop iteration...